photon-core/src/main/java/org/photonvision/vision/processes/VisionSourceManager.java

```diff
@@ -528,80 +205,62 @@ public void setIgnoredCamerasRegex(String ignoredCamerasRegex) {
-                // If cscore hasnt passed this other paths aka a path by id or a path as in usb port then we
-                // cant guarantee it is a valid camera.
+                    && device.type() == CameraType.UsbCamera) {
+                    logger.trace(
+                            "Skipping device with no other paths: \""
+                                    + device.name()
+                                    + "\" at \""
+                                    + device.path());
+                } else if (Arrays.stream(usbDevice.otherPaths).anyMatch(it -> it.contains("csi-video"))
+                            || usbDevice.name().equals("unicam")) {
+                    logger.trace(
+                            "Skipping CSI device from CSCore: \""
+                                    + device.name()
+                                    + "\" at \""
+                                    + device.path()
+                                    + "\"");
+                } else {
+                    valid = true;
+                }
             } else {
+                valid = true;
+            }
+            if (valid) {
                 filteredDevices.add(device);
-                logger.trace(
-                        "Adding local video device - \"" + device.name + "\" at \"" + device.path + "\"");
+                logger.trace("Adding local video device - \"" + device.name() + "\" at \"" + device.path() + "\"");
             }
         }
         return filteredDevices;
     }
 
-    private static List<VisionSource> loadVisionSourcesFromCamConfigs(
-            List<CameraConfiguration> camConfigs, boolean createSources) {
-        var cameraSources = new ArrayList<VisionSource>();
-        for (var configuration : camConfigs) {
-            // In unit tests, create dummy
-            if (!createSources) {
-                cameraSources.add(new TestSource(configuration));
-                continue;
-            }
-
-            boolean is_pi = Platform.isRaspberryPi();
+    private static Optional<VisionSource> loadVisionSourceFromCamConfig(CameraConfiguration configuration) {
+        VisionSource source = null;
+        boolean is_pi = Platform.isRaspberryPi();
 
-            if (configuration.cameraType == CameraType.ZeroCopyPicam && is_pi) {
-                // If the camera was loaded from libcamera then create its source using libcamera.
-                var piCamSrc = new LibcameraGpuSource(configuration);
-                cameraSources.add(piCamSrc);
-            } else {
-                var newCam = new USBCameraSource(configuration);
-                if (!newCam.getCameraQuirks().hasQuirk(CameraQuirk.CompletelyBroken)
-                        && !newCam.getSettables().videoModes.isEmpty()) {
-                    cameraSources.add(newCam);
-                }
+        if (configuration.cameraType == CameraType.ZeroCopyPicam && is_pi) {
```

Comment:
can cameraType ever be picamera, but is_pi be false?

Reply:
I'm unsure, this was just moved old code